tests/storage/restore.go

```diff
@@ -1366,6 +1429,38 @@ var _ = SIGDescribe("VirtualMachineRestore Tests", func() {
 				Entry("to a new VM", true),
 			)
 
+			DescribeTable("[Serial]Should restore a vm with backend storage", Serial, func(restoreToNewVM, onlineSnapshot bool) {
+				By("Setting the VMState storage class to snapshot")
+				kv := libkubevirt.GetCurrentKv(virtClient)
+				kv.Spec.Configuration.VMStateStorageClass = snapshotStorageClass
+				config.UpdateKubeVirtConfigValueAndWait(kv.Spec.Configuration)
+
+				vm = createVMWithCloudInit(cd.ContainerDiskFedoraTestTooling, snapshotStorageClass)
+				vm.Spec.Template.Spec.Domain.Devices.TPM = &v1.TPMDevice{Persistent: pointer.P(true)}
+				vm, vmi = createAndStartVM(vm)
+				Eventually(ThisVM(vm)).WithTimeout(300 * time.Second).WithPolling(time.Second).Should(BeReady())
+
+				By("Expecting the creation of a backend storage PVC with the right storage class")
+				pvcs, err := virtClient.CoreV1().PersistentVolumeClaims(vmi.Namespace).List(context.Background(), metav1.ListOptions{
+					LabelSelector: "persistent-state-for=" + vmi.Name,
+				})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(pvcs.Items).To(HaveLen(1))
+
+				doRestoreNoVMStart("", console.LoginToFedora, onlineSnapshot, true, getTargetVMName(restoreToNewVM, newVmName))
+				startVMAfterRestore(getTargetVMName(restoreToNewVM, newVmName), "", true, console.LoginToFedora)
+				Expect(restore.Status.Restores).To(HaveLen(2))
+
+				if restoreToNewVM {
+					checkNewVMEquality()
+				}
+			},
+				Entry("to the same VM with online snapshot", false, true),
```

Comment:
I thought we wanted to avoid triggering errors/failing because could unexpectedly prevent users who already have VMs with backend storage from creating a snapshot. Currently the snapshot just happens like we did before; regular volumes are snapshotted and the backend PVC is ignored.

Reply:
That's right, should I add the webhooks back? We discussed this some weeks ago and decided that controlling/blocking this in the controller was a good idea. If the webhooks are ok I can add them back and avoid having all this code checking for online VM/clone in the controllers.

Reply:
I will definitely add the webhook back for clone. For an online snapshot, you can decide whether it is better to block/avoid the operation in the controller or in a webhook. I'll continue with this once I come back from holidays, my plan is to open the online snapshot PR short after this is merged so don't want to focus too much on blocking stuff in the controller.